File: src/main/scala/com/seu/geomesa/scala/spark/CoreDemo.scala
package com.seu.geomesa.scala.spark

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.hbase.io.ImmutableBytesWritable
import org.apache.spark.SparkConf
import org.apache.spark.sql.SparkSession
import org.geotools.data.Query
import org.geotools.feature.simple.{SimpleFeatureImpl, SimpleFeatureTypeImpl}
import org.geotools.filter.text.ecql.ECQL
import org.locationtech.geomesa.spark.SpatialRDD.toGeoJSONString
import org.locationtech.geomesa.spark.{GeoMesaSpark, GeoMesaSparkKryoRegistrator}

import java.util

object CoreDemo {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.registerKryoClasses(Array(classOf[ImmutableBytesWritable]))
    val spark = SparkSession.builder().appName("JtsDemo").master("local[*]")
      .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
      .config("spark.kryo.registrator", classOf[GeoMesaSparkKryoRegistrator].getName)
      .config(conf)
      .getOrCreate()
    // DataStore params to a hypothetical GeoMesa Accumulo table
    val dsParams = Map(
      "hbase.zookeepers" -> "zookeeper",
      "hbase.catalog" -> "higeo:st_gdelt")
    val map = new util.HashMap[String, String]()
    dsParams.foreach(kv => map.put(kv._1, kv._2))


    // create RDD with a geospatial query using GeoMesa functions
    val spatialRDDProvider = GeoMesaSpark(map)
    val filter = ECQL.toFilter("CONTAINS(POLYGON((0 0, 0 90, 90 90, 90 0, 0 0)), geom)")
    val query = new Query("gdelt", filter)
    val resultRDD = spatialRDDProvider.rdd(new Configuration, spark.sparkContext, dsParams, query)

    println(resultRDD.count())
    resultRDD.take(10).foreach(println)
  }
}

File: src/main/java/com/seu/geomesa/spark/core/CoreDemo.java
package com.seu.geomesa.spark.core;

import com.seu.utils.ScalaUtil;
import org.apache.hadoop.conf.Configuration;
import org.apache.spark.sql.SparkSession;
import org.geotools.data.Query;
import org.geotools.filter.text.cql2.CQLException;
import org.geotools.filter.text.ecql.ECQL;
import org.locationtech.geomesa.spark.GeoMesaSpark;
import org.locationtech.geomesa.spark.SpatialRDD;
import org.locationtech.geomesa.spark.SpatialRDDProvider;
import org.opengis.filter.Filter;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class CoreDemo {
    public static void main(String[] args) throws CQLException {
        SparkSession spark = SparkSession.builder().appName("GeoMesa Spark Core Demo").master("local[*]").getOrCreate();

        Map<String, String> params = new HashMap<>();
        params.put("hbasehbase.zookeepers", "zookeeper");
        params.put("hbase.catalog", "higeo:st_gdelt");
        scala.collection.immutable.Map<String, String> map = ScalaUtil.toScalaImmutableMap(params);

        SpatialRDDProvider spatialRDDProvider = GeoMesaSpark.apply(params);
        Filter filter = ECQL.toFilter("CONTAINS(POLYGON((0 0, 0 90, 90 90, 90 0, 0 0)), geom)");
        Query query = new Query("gdelt", filter);

        SpatialRDD spatialRDD = spatialRDDProvider.rdd(new Configuration(), spark.sparkContext(), map, query);
        System.out.println(spatialRDD.count());
    }
}
